src/app/info/InfoPage.tsx

```diff
@@ -1,47 +1,41 @@
-"use client";
-
 import React, { useState, useEffect } from "react";
-import WikiItems from "@/app/info/WikiItems";
+import WikiItem from "@/app/info/WikiItem";
 
 import { Supabase } from "@/supabaseUtils";
 import { DbWikiRow } from "@/databaseUtils";
 import supabase from "@/supabaseClient";
-import Error from "next/error";
 
-interface QueryType {
-  data: DbWikiRow | null;
-  error: Error | null;
+import { PostgrestError} from "@supabase/supabase-js";
+
+interface QuerySuccessType {
+  data: DbWikiRow;
+  error: null;
+}
+interface QueryFailureType {  
+  data: null
+  error: PostgrestError
 }
 
+type QueryType = QuerySuccessType | QueryFailureType
+
 const getTopWikiRow = async (supabase: Supabase): Promise<QueryType> => {
     const query = (await supabase.from("wiki").select("*").limit(1).single()) as QueryType;
     return query;
 };
 
-const InfoPage: React.FC = () => {
-    const [topRow, setTopRow] = useState<DbWikiRow | null>(null); 
-    try {
-      useEffect(() => {
-        const fetchData: () => Promise<void> = async () => {
-            const temp: QueryType = await getTopWikiRow(supabase);
-            if (temp.error) {
-              throw temp.error;
-            } else if(temp.data) {
-              setTopRow(temp.data);
-            }
-        };
-        fetchData();
-      }, []);
-    } catch (e: any) {
-      return (
-      <>
-        <h3>Error Occured</h3>
-        <p>{e}</p>
-      </>
-      );
-    }
-
-    return (topRow && topRow.order !== -1) ? <WikiItems row={topRow}/> : <></>;
+export async function getServerSideProps() {
+  const query: QueryType = await getTopWikiRow(supabase);
+  return query
+}
```

Comment:
It looks like you're trying to use [server side props like this](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props), but I think there's a few implementation details we need to iron out. Lets have a chat about it :)